### PR TITLE
NCache Managed ARM template which creates cluster using script

### PR DESCRIPTION
**Pull request by mohsinnasir**
NCache Managed arm template
This template creates N number of NCache 5.0 SP2 azure market place image and using the custom script extension creates the cluster of a cache for the user
Using this arm template user can also activate their servers by providing licensing information

### CODE REVIEW

**Comment by mohsinnasir** [comment]
I am unable to understand why the sample folder was not deployed as when i tried to open that link it was successful
Kindly look into it

**Comment by bmoore-msft** [comment]
you're missing the metadata.json file

File: ncache-vm-cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# NCache Manged ARM Template
```
Comment (bmoore-msft) [inline ncache-vm-cluster/README.md:1]: need to add the badges and buttons, see the pipeline task for a snippet (or the contribution guide)

File: ncache-vm-cluster/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+### SubnetPrefix
+Subet address prefix in VNET
+
+### VirtualNetworkName
+VNET Name
+
+### NCacheClusterCreationScriptFileUri
+URI of Blob storage of "NCacheConfiguration.ps1" file. Which will be uploaded by the user before deploying the template. This Script will create a NCache Cluster and is mandatory to upload.
+
```
Comment (bmoore-msft) [inline ncache-vm-cluster/README.md:96]: remove extraneous blank lines

File: ncache-vm-cluster/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+    "numberOfVMs": {
+      "type": "int",
+      "minValue": 1,
+      "defaultValue": 2,
+      "metadata": {
+        "description": "Number of Azure Machines to be deployed"
+      }
+    },
+    "licenseKey": {
+      "type": "string",
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.json:14]: is this a secret?  if so, make it a secure string

File: ncache-vm-cluster/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+      "defaultValue": "NCacheVM"
+    },
+    "virtualMachineSize": {
+      "type": "string",
+      "defaultValue": "Standard_DS2_v2"
+    },
+    "adminUsername": {
+      "type": "string",
+      "defaultValue": "azadmin"
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.json:124]: remove defaultValue

File: ncache-vm-cluster/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "defaultValue": "123Islamabad"
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.json:128]: remove defaultValue

File: ncache-vm-cluster/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+    "NCacheClusterCreationScriptFileUri": {
+      "type": "string",
+      "defaultValue": "NotSpecified"
+    }
+  },
+  "resources": [
+    {
+      "name": "[concat(parameters('virtualMachineNamePrefix'), copyIndex(1))]",
+      "type": "Microsoft.Compute/virtualMachines",
+      "apiVersion": "2017-12-01",
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.json:163]: update apiVersions throughout - nothing more than 2 years old (unless it's the latest)

File: ncache-vm-cluster/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+      }
+    },
+    {
+      "name": "[concat(parameters('networkInterfaceName'), copyIndex(1))]",
+      "type": "Microsoft.Network/networkInterfaces",
+      "apiVersion": "2017-11-01",
+      "location": "[parameters('location')]",
+      "copy": {
+        "name": "networkInterfaceLoop",
+        "mode": "serial",
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.json:239]: this shouldn't need to be serial - is there a particular reason for it?

File: ncache-vm-cluster/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,401 @@
...
+      "name": "[concat(parameters('publicIpAddressName'), copyIndex(1))]",
+      "type": "Microsoft.Network/publicIpAddresses",
+      "apiVersion": "2017-11-01",
+      "location": "[parameters('location')]",
+      "properties": {
+        "publicIpAllocationMethod": "Dynamic"
+      },
+      "copy": {
+        "name": "ipLoop",
+        "mode": "serial",
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.json:277]: this doesn't need to be serial - remove or change to parallel

File: ncache-vm-cluster/azuredeploy.parameters.json [hunk 9]
```diff
@@ -0,0 +1,67 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.parameters.json:4]: remove parameters for which you have defaultValues

File: ncache-vm-cluster/azuredeploy.parameters.json [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+      "value": "NotSpecified"
+    },
+    "virtualMachineNamePrefix": {
+      "value": "NCacheVM"
+    },
+    "virtualMachineSize": {
+      "value": "Standard_DS2_v2"
+    },
+    "adminUsername": {
+      "value": "ncadmin"
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.parameters.json:45]: GEN-UNIQUE

File: ncache-vm-cluster/azuredeploy.parameters.json [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    "adminPassword": {
+      "value": "12345Alachisoft"
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.parameters.json:48]: GEN-PASSWORD

File: ncache-vm-cluster/azuredeploy.parameters.json [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+      "value": "Subnet1"
+    },
+    "subnetPrefix": {
+      "value": "10.0.0.0/24"
+    },
+    "virtualNetworkName": {
+      "value": "MyVNET1"
+    },
+    "NCacheClusterCreationScriptFileUri": {
+      "value": "https://yourstorageaccount.blob.core.windows.net/tourcontainer/NCacheConfiguration.ps1"
```
Comment (bmoore-msft) [inline ncache-vm-cluster/azuredeploy.parameters.json:63]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts

File: ncache-vm-cluster/metadata.json [hunk 13]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (bmoore-msft) [inline ncache-vm-cluster/metadata.json:1]: if this is not supported in Azure US Government, add the environments property to select clouds that are supported

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**